src/Network/Readability/Shortener.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Network.Readability.Shortener
+    ( Meta(..)
+    , ShortenerResponse(..)
+    , shortenUrl
+    , retrieveUrl
+    ) where
+
+import Control.Applicative ((<$>))
+
+import qualified Data.ByteString.Char8 as BS
+
+import Data.Text (Text)
+import Data.Aeson (FromJSON, decode)
+import Data.Aeson.TH (deriveFromJSON, defaultOptions, fieldLabelModifier)
+
+import Network.HTTP.Conduit --(parseUrl, responseBody, setQueryString, withManager, httpLbs)
+
+import Network.Readability.Parser (Article(..))
+
+
+data Meta = Meta
+    { meta_article :: Maybe Article
+    , meta_url :: Maybe Text -- Url
+    , meta_rdd_url :: Text -- Url
+    , meta_id :: Text
+    , meta_full_url :: Maybe Text
+    } deriving (Show)
+
+data ShortenerResponse = ShortenerResponse
+    { meta :: Meta
+    , messages :: [Text]
+    , success :: Bool
+    } deriving (Show)
+
+$(deriveFromJSON defaultOptions{ fieldLabelModifier = drop (length ("meta_" :: String)) } ''Meta)
+$(deriveFromJSON defaultOptions ''ShortenerResponse)
+
+apiPrefix :: String
+apiPrefix = "https://readability.com/api/shortener/v1/urls"
+
+shortenUrl :: String -> IO (Maybe ShortenerResponse)
+shortenUrl source_url  = shortenUrlRequest $ BS.pack source_url
+
+shortenUrlRequest :: FromJSON a => BS.ByteString -> IO (Maybe a)
```

Comment:
In reality, shortenerUrlRequest doesn't return any FromJSON but ShortenerResponse